libcontainer/cgroups/fs/memory.go

```diff
@@ -108,28 +121,28 @@ func setMemoryAndSwap(path string, cgroup *configs.Cgroup) error {
 		// for memory and swap memory, so it won't fail because the new
 		// value and the old value don't fit kernel's validation.
 		if memoryUsage.Limit < uint64(cgroup.Resources.MemorySwap) {
```

Comment:
when reset memory swap to -1 while the container still have memory.limit and memory.memsw.limit, this will fail and we will set memory limit to -1 while memsw is still in place. I think that will cause an error.
We should check if we want to reset memory / memoryswap here and change the order accordingly.